sbi/inference/posteriors/mcmc_posterior.py

```diff
@@ -352,9 +352,90 @@ def sample(
                 raise NameError(f"The sampling method {method} is not implemented!")
 
         samples = self.theta_transform.inv(transformed_samples)
+        samples = samples.reshape((*sample_shape, -1))  # type: ignore
```

Comment:
This actually is a duplicate. In the sample function, I will add comments to sample_batched

Reply:
done

Reply:
This actually was a duplicate reshape in the sample function, I added comments to sample_batched

Reply:
Hm, I am still confused by this reshape. Why do we have to do this in the original `sample` method. How was it changed? And it seems we are squeezing all potential event dimensions of the samples in the last one? What am I missing?

Reply:
I'm unsure if I added it or if this is a leftover from Michael. This will recover the sample_shape (which before probably was missing?). 

I agree, that this will squeeze all event dimensions in the last dimension. As far as I can see all samplers (pyro,pymc) do **require** such a squeeze and will also return it. I don't think we currently test for any parameters with > 1 ndim, so checking and adapting this for all posterior would be another issue. 

I would keep the reshape, as it makes the sample_shape consistent (which without would also be squeezed in one dimension).

Reply:
this is resolved now and move to the batched MCMC PR?

Reply:
Yes, resolved for this PR!